packages/intrepid2/src/Cell/Intrepid2_CellToolsDefControlVolume.hpp

```diff
@@ -394,37 +395,33 @@ namespace Intrepid2 {
 
     // construct edge and face map for the cell type
     const ordinal_type numEdge = primaryCell.getSubcellCount(1);
-    Kokkos::View<ordinal_type**,Kokkos::LayoutRight,Kokkos::HostSpace> edgeMapHost("CellTools::getSubcvCoords::edgeMapHost", numEdge, 3);
+    Kokkos::View<ordinal_type**,deviceType> edgeMap("CellTools::getSubcvCoords::edgeMap", numEdge, 3);
+    auto edgeMapHost = Kokkos::create_mirror_view(edgeMap);
     for (ordinal_type i=0;i<numEdge;++i) {
       edgeMapHost(i,0) = primaryCell.getNodeCount(1, i);
       for (ordinal_type j=0;j<edgeMapHost(i,0);++j)
         edgeMapHost(i,j+1) = primaryCell.getNodeMap(1, i, j);
     }
 
     const ordinal_type numFace = (spaceDim > 2 ? primaryCell.getSubcellCount(2) : 0);
-    Kokkos::View<ordinal_type**,Kokkos::LayoutRight,Kokkos::HostSpace> faceMapHost("CellTools::getSubcvCoords::faceMapHost", numFace, 5);
+    Kokkos::View<ordinal_type**,deviceType> faceMap("CellTools::getSubcvCoords::faceMap", numFace, 5);
+    auto faceMapHost = Kokkos::create_mirror_view(faceMap);
     for (ordinal_type i=0;i<numFace;++i) {
       faceMapHost(i,0) = primaryCell.getNodeCount(2, i);
       for (ordinal_type j=0;j<faceMapHost(i,0);++j)
         faceMapHost(i,j+1) = primaryCell.getNodeMap(2, i, j);
     }
 
-    // create mirror to device
-    auto edgeMap = Kokkos::create_mirror_view(typename SpT::memory_space(), edgeMapHost);
-    auto faceMap = Kokkos::create_mirror_view(typename SpT::memory_space(), faceMapHost);
-
     Kokkos::deep_copy(edgeMap, edgeMapHost);
     Kokkos::deep_copy(faceMap, faceMapHost);
 
     // parallel run
-    typedef Kokkos::DynRankView<subcvCoordValueType,subcvCoordProperties...> subcvCoordViewType;
-    typedef Kokkos::DynRankView<cellCoordValueType,cellCoordProperties...>   cellCoordViewType;
-    typedef Kokkos::View<ordinal_type**,Kokkos::LayoutRight,SpT>             mapViewType;
-
-    typedef typename ExecSpace<typename subcvCoordViewType::execution_space,SpT>::ExecSpaceType ExecutionSpace;
+    using subcvCoordViewType = decltype(subcvCoords);
+    using cellCoordViewType = decltype(cellCoords);
+    using mapViewType = decltype(edgeMap);
 
     const auto loopSize = numCells;
-    Kokkos::RangePolicy<ExecutionSpace,Kokkos::Schedule<Kokkos::Static> > policy(0, loopSize);
+    Kokkos::RangePolicy<typename deviceType::execution_space,Kokkos::Schedule<Kokkos::Static> > policy(0, loopSize);
```

Comment:
using view execution space means that you assume that this is light weight computation. this still assumes that input views are accessible but it does not check its accessibilities.